store/sqlstore/shared_channel_store.go

```diff
@@ -386,24 +386,35 @@ func (s SqlSharedChannelStore) GetRemoteByIds(channelId string, remoteId string)
 }
 
 // GetRemotes fetches all shared channel remotes associated with channel_id.
-func (s SqlSharedChannelStore) GetRemotes(channelId string) ([]*model.SharedChannelRemote, error) {
+func (s SqlSharedChannelStore) GetRemotes(opts model.SharedChannelRemoteFilterOpts) ([]*model.SharedChannelRemote, error) {
 	var remotes []*model.SharedChannelRemote
 
 	query := s.getQueryBuilder().
 		Select("*").
-		From("SharedChannelRemotes").
-		Where(sq.Eq{"SharedChannelRemotes.ChannelId": channelId})
+		From("SharedChannelRemotes")
+
+	if opts.ChannelId != "" {
+		query = query.Where(sq.Eq{"SharedChannelRemotes.ChannelId": opts.ChannelId})
+	}
+
+	if opts.RemoteId != "" {
+		query = query.Where(sq.Eq{"SharedChannelRemotes.RemoteId": opts.RemoteId})
+	}
+
+	if !opts.InclUnconfirmed {
+		query = query.Where(sq.Eq{"SharedChannelRemotes.IsInviteConfirmed": true})
+	}
```

Comment:
Do we need to repeat the table name?

Reply:
The field names are not ambiguous so I'll remove the table name.

Reply:
Fixed via https://github.com/mattermost/mattermost-server/pull/17049/commits/9c0c3116b0f6e8a6d44f6ea759397a01ba985ef8